crates/astria-conductor/src/executor.rs

```diff
@@ -114,6 +118,12 @@ struct Executor<C> {
     /// so that we can mark the block as final on the execution layer when
     /// we receive a finalized sequencer block.
     sequencer_hash_to_execution_hash: HashMap<Vec<u8>, Vec<u8>>,
+
+    /// most recently executed sequencer block hash
+    most_recently_executed_block_hash: Hash,
```

Comment:
shouldn't be a tendermint hash as it isn't a tendermint hash, it's a rollup block hash (which may not use the same algorithm) 

you can replace `execution_state` with this